test/unit/index.js

```diff
@@ -0,0 +1,155 @@
+const test = require('tape')
+
+const TokenTracker = require('../../lib/')
+const { setupSimpleTokenEnvironment } = require('../helper')
+
+test('tracker with minimal options', async function (t) {
+  t.plan(1)
+  let tracker
+  try {
+    const { provider } = await setupSimpleTokenEnvironment()
+    tracker = new TokenTracker({ provider })
+    t.deepEqual(tracker.serialize(), [], 'should serialize as an empty array')
+    t.end()
+  } finally {
+    tracker.stop()
+  }
+})
+
+test('tracker with minimal token', async function (t) {
+  t.plan(2)
+  let tracker
+  try {
+    const { addresses, provider, tokenAddress } = await setupSimpleTokenEnvironment()
+    tracker = new TokenTracker({
+      pollingInterval: 20,
+      provider,
+      tokens: [
+        {
+          address: tokenAddress,
+        }
+      ],
+      userAddress: addresses[1],
+    })
+
+    const updates = []
+    tracker.on('update', (tokens) => {
+      updates.push(tokens)
+    })
+
+    await new Promise((resolve) => setTimeout(() => resolve(), 200))
```

Comment:
Can we think of a way to avoid these 200ms waits?

Reply:
I think the `setTimeout` is required, but I don't think it has to be 200ms. I'll experiment with smaller values.

Reply:
It does not seem to work with smaller timeouts. `150ms` is the lowest value I tested that worked, but that's hardly worth the reduction (especially since it might result in failure on a slower machine)

Reply:
Oh, I more meant is there a way to avoid them altogether—if there isn't a good way currently we can leave it with a larger number, as that'll be less (?) flaky

Reply:
They're required because testrpc will do a bunch of async things when it receives a transaction, then the block tracker will regularly poll for new blocks. Even if I make the polling time extremely small/zero, the async work they're doing seems to take around this long to get through. I don't know if a way to explicitly wait unfortunately.

Maybe if we inject a block tracker, instead of using the internal one? 🤔 Then we could wait for the next block. But still I don't think it'd speed things up a great deal - it'd just be more reliable (in theory).